js/piwik.js

```diff
@@ -5759,9 +5762,30 @@ if (typeof window.Matomo !== 'object') {
              * @param bool
              */
             this.setSecureCookie = function (enable) {
+                if(location.protocol !== 'https:') {
+                    logConsoleError("Error in setSecureCookie: You cannot use `Secure` on http.");
+                }
                 configCookieIsSecure = enable;
             };
 
+            /**
+             * Set the SameSite attribute for cookies to a custom value.
+             * You might want to use this if your site is running in an iframe since
+             * then it will only be able to access the cookies if SameSite is set to 'None'.
+             * Sets to Lax if invalid parameter is passed, sets CookieIsSecure to true on None.
```

Comment:
could we mention that `None` will only work on HTTPS?

Reply:
btw thinking we maybe could also mention that `None` might cause issues if the user is sometimes on HTTPS and sometimes not (eg a site allows access through HTTP and HTTPS).

Reply:
also thinking we might need to mention that setting `None` will also directly set `secure cookie flag` and if users browse HTTP and HTTPS during same visits it would create two visits because two different cookies would be created AFAIK

Reply:
Along the lines of this?
setSecureCookie
"Warning: If your site is available under http and https, setting this might lead to duplicate or incomplete visits."

setCookieSameSite
"Warning: Sets CookieIsSecure to true on None, because None will only work with Secure; cookies If your site is available under http and https, using "None" might lead to duplicate or incomplete visits."